src/date-picker/index.tsx

```diff
@@ -168,11 +168,6 @@ const DatePicker = React.forwardRef(
 
     baseProps.className = clsx(baseProps.className, styles.root, styles['date-picker-container']);
 
-    const handleMouseDown = (event: React.MouseEvent) => {
```

Comment:
This was preventing text selection in date picker even after removing `user-select: none` from the dropdown.

It was introduced 3.5 years ago in `CR-49510605` with the justification to prevent an undesired focus ring from appearing when clicking the calendar, but I can't reproduce this today.